src/core/memblock.h

```diff
@@ -0,0 +1,17 @@
+#ifndef CORE_MEM_BLOCK_H
+#define CORE_MEM_BLOCK_H
+
+#include <stdint.h>
+
+typedef struct {
+	char *mem;
+	uint32_t size;
+} memblock;
+
+int core_memblock_init(memblock *block, uint32_t initial_size);
+
+int core_memblock_check_size(memblock *block, uint32_t size);
```

Comment:
What about `core_memory_block_fit_bytes` here instead?

Reply:
I went ahead and changed it but I'm not thrilled with the new name. To me fit_bytes implies that it will reduce the size of the memory block to the value passed in when in fact it will only grow the memory block.

Reply:
You're right but check size also only seems to check if the size fits, and not the it will actually increase the size if needed.

I'm open to a different name, but I couldn't think of a better one.

Maybe `hold_bytes`? No idea..

Reply:
ensure_size?